WordPress/src/main/java/org/wordpress/android/ui/mlp/CategoriesItemDecoration.kt

```diff
@@ -0,0 +1,38 @@
+package org.wordpress.android.ui.mlp
+
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.view.LayoutInflater
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import org.wordpress.android.R
+import org.wordpress.android.ui.mlp.ModalLayoutPickerListItem.ViewType
+import kotlin.math.max
+
+/**
+ * Implements the Categories 'sticky' row behavior
+ */
+class CategoriesItemDecoration : RecyclerView.ItemDecoration() {
+    private lateinit var stickyView: View
+    private lateinit var scrollingView: View
+
+    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
+        super.getItemOffsets(outRect, view, parent, state)
+        if (ViewType.CATEGORIES.id == parent.adapter?.getItemViewType(parent.getChildAdapterPosition(view))) {
+            scrollingView = view
+        }
+    }
+
+    override fun onDrawOver(canvas: Canvas, parent: RecyclerView, state: RecyclerView.State) {
+        super.onDrawOver(canvas, parent, state)
+        if (scrollingView.y > 0 || !::scrollingView.isInitialized) return
+        if (!::stickyView.isInitialized) {
+            stickyView = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.modal_layout_picker_categories_row, parent, false)
+        }
+        canvas.save()
+        canvas.translate(0f, max(0f, stickyView.top.toFloat() - stickyView.height.toFloat()))
+        scrollingView.draw(canvas)
+        canvas.restore()
+    }
+}
```

Comment:
This approach on implementing the categories 'sticky' header may be revisited since i has some limitations regarding handling touches. This become apparent while [implementing the categories selection in a next PR](https://github.com/wordpress-mobile/WordPress-Android/pull/12573). A workaround and another approach on this is examined.